### PR TITLE
Add filters to career API

### PR DESCRIPTION
**Pull request by oleast**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

## Description of changes

Add API filters so that searches can be done against the database instead of just in the front-end.

### CODE REVIEW

File: apps/careeropportunity/filters.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from django_filters import filters, filterset
+from watson import search as watson_search
+
+from .models import CareerOpportunity
+
+
+class CharInFilter(filters.BaseInFilter, filters.CharFilter):
+    pass
```
Comment (FluidSense) [inline apps/careeropportunity/filters.py:8]: Is this necessary?

Comment (oleast) [reply apps/careeropportunity/filters.py:8]: Yes, it is kinda strange but it essentially means: _combine these classes together, without adding anything else_

Comment (FluidSense) [reply apps/careeropportunity/filters.py:8]: Aha! I see it now, thanks!

**Comment by FluidSense** [comment]
@oleast Could you make an example of a request to test this with? 😄

**Pull request merged**